File: deploy_tools/deploy.js
const Client = require('ssh2-sftp-client');
const sftp = new Client();
const path = require('path');

const config = {
  host: '46.202.183.28',
  port: 65002,
  username: 'u425316205',
  password: 'Hisab@20026'
};

const filesToUpload = [
    'app/Http/Controllers/Admin/ChatbotController.php',
    'app/Models/ChatbotLog.php',
    'database/migrations/2026_08_31_000001_create_chatbot_logs_table.php',
    'public/css/chatbot.css',
    'public/js/chatbot.js',
    'resources/views/components/chatbot-widget.blade.php',
    'resources/views/layouts/admin.blade.php',
    'resources/views/layouts/app.blade.php',
    'resources/views/layouts/employee.blade.php',
    'routes/web.php'
];

async function deploy() {
  try {
    await sftp.connect(config);
    const paths = [
        '/home/u425316205/domains/hisabmittra.in/public_html',
        '/home/u425316205/domains/hisabmittra.in/public_html/crm'
    ];
    
    for (const remoteBasePath of paths) {
        console.log('\\n--- Deploying to ' + remoteBasePath + ' ---');
        for (const file of filesToUpload) {
            const localFile = path.join(__dirname, '../', file);
            const remoteFile = remoteBasePath + '/' + file;
            
            const remoteDir = path.dirname(remoteFile);
            const exists = await sftp.exists(remoteDir);
            if (!exists) {
                console.log('Creating ' + remoteDir);
                await sftp.mkdir(remoteDir, true);
            }
            
            console.log('Uploading ' + file + ' ...');
            await sftp.put(localFile, remoteFile);
        }
        
        const migratorScript = `<?php
        require __DIR__.'/../vendor/autoload.php';
        $app = require_once __DIR__.'/../bootstrap/app.php';
        $kernel = $app->make(Illuminate\\Contracts\\Http\\Kernel::class);
        $response = $kernel->handle(
            $request = Illuminate\\Http\\Request::capture()
        );
        Illuminate\\Support\\Facades\\Artisan::call('migrate', ['--force' => true]);
        echo "Done! " . Illuminate\\Support\\Facades\\Artisan::output();
        `;
        
        const localMigrator = path.join(__dirname, 'migrator.php');
        require('fs').writeFileSync(localMigrator, migratorScript);
        
        await sftp.put(localMigrator, remoteBasePath + '/public/migrator.php');
    }
    
    console.log('Deployment completely finished for both folders!');
  } catch (err) {
    console.error('Deployment error: ', err.message);
  } finally {
    sftp.end();
  }
}

deploy();

File: deploy.py
import paramiko
import os

host = "46.202.183.28"
port = 65002
username = "u425316205"
password = "Hisab@20026"

ssh = paramiko.SSHClient()
ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
ssh.connect(host, port, username, password)

sftp = ssh.open_sftp()
base_remote = "domains/hisabmittra.in/public_html/crm"

files = [
    'app/Http/Controllers/Admin/TaskController.php',
    'app/Http/Controllers/Admin/EmployeeController.php',
    'app/Http/Controllers/Employee/TaskController.php',
    'app/Http/Controllers/Employee/CustomerController.php',
    'resources/views/admin/tasks/index.blade.php',
    'resources/views/admin/employees/index.blade.php',
    'resources/views/employee/tasks/index.blade.php',
    'resources/views/employee/tasks/show.blade.php',
    'app/Models/Customer.php',
    'resources/views/admin/customers/show.blade.php',
    'resources/views/admin/reports/index.blade.php',
    'resources/views/employee/customers/show.blade.php',
    'resources/views/components/topbar.blade.php'
]

for f in files:
    local_path = f.replace('/', '\\')
    remote_path = base_remote + '/' + f
    
    # ensure remote dir exists
    remote_dir = os.path.dirname(remote_path)
    stdin, stdout, stderr = ssh.exec_command(f"mkdir -p {remote_dir}")
    stdout.read()
        
    print(f"Uploading {f}...")
    sftp.put(local_path, remote_path)

sftp.close()

print("Running migrations and clearing cache on server...")
commands = [
    "cd domains/hisabmittra.in/public_html/crm",
    "php artisan migrate --force",
    "php artisan route:clear",
    "php artisan view:clear"
]
stdin, stdout, stderr = ssh.exec_command(" && ".join(commands))
print(stdout.read().decode())
print(stderr.read().decode())
ssh.close()
print("Deployment completed successfully!")
